src/lightning/app/runners/cloud.py

```diff
@@ -423,6 +435,17 @@ def _resolve_root(self) -> Path:
             root = root.parent
         return root
 
+    def _resolve_env_root(self) -> Path:
+        """Determine whether the root of environment sync files exists."""
+        root = Path(SYS_CUSTOMIZATIONS_SYNC_ROOT)
+        if root.exists():
+            return root
+        return None
+
+    def _prepare_sys_customizations_sync(self, sys_customizations_root: Path, root: Path) -> None:
+        path_to_sync = Path(root, SYS_CUSTOMIZATIONS_SYNC_PATH)
+        shutil.copytree(sys_customizations_root, path_to_sync, dirs_exist_ok=True)
```

Comment:
I think we have another op. in the source code upload part which copies everything to a temp directory for tarring. Maybe it would be better to copy these files directly in there? The class is in `lightning.app.source_code.local` :smiley:

Reply:
Yes, we can! I was thinking here it would be good to have it as a CloudSpace-specific thing... So not sure we should always do this, we can put a bit of logic into `lightning.app.source_code.local` and call it only for CloudSpaces. 
[Since the change is already included to `plugin-server` to provide basic sync, I'll back to that PR after base-images and BE PRs merge.]

Reply:
Ah I see, it's only used in the cloudspace_dispatch - yeah that's fine :smiley: